File: updated_code/code/java/136-single-number.java
package com.example;

/**
 * This class provides a solution to the problem of finding a single number that appears only once in a non-empty array of integers, where every element appears twice except for one.
 */
public class SingleNumber {

    /**
     * Method to find the single number in an array of integers.
     *
     * @param nums a non-empty array of integers, where every element appears twice except for one.
     * @return the single number that appears only once in the array.
     */
    public int singleNumber(int[] nums) {
        int result = 0;
        for (int num : nums) {
            result ^= num;
        }
        return result;
    }
}

File: updated_code/code/python/136-single-number.py
from typing import List

class Solution:
    """
    This class is designed to solve the problem of finding the single number in an array where every element appears twice except for one.
    """
    """
    This method uses the XOR operation to find the single number in the array.
    The XOR operation has the property that a ^ a = 0 and any number a ^ 0 = a.
    So, when we XOR all the numbers in the array, the numbers that appear twice will cancel each other out, leaving the single number.
    """
    def single_number(self, nums: List[int]) -> int:
        """
        This function finds the single number in the array.
        It works by XORing all the numbers in the array.
        """
        result = 0
        for num in nums:
            result ^= num
        return result